src/main/java/reactor/rabbitmq/Sender.java

```diff
@@ -207,19 +207,61 @@ protected Scheduler createScheduler(String name) {
      * @see SendOptions#trackReturned(boolean)
      */
     public Flux<OutboundMessageResult> sendWithPublishConfirms(Publisher<OutboundMessage> messages, SendOptions options) {
+        return sendWithCorrelatedPublishConfirms(Flux.from(messages).map(message -> (OutboundMessage<Object>) message), options)
+            .map(Function.identity());
```

Comment:
This is the only part of the changes that are somewhat squint-worthy, but necessary to make the compiler happy

Reply:
the suggestions above will remove the need for such casts, but you'll probably need 2 separate operator implementations, or at least a bit more juggling at the operator level